examples/with-sentry/components/withSentry.js

```diff
@@ -0,0 +1,35 @@
+import React from 'react'
+import Raven from 'raven-js'
+
+const SENTRY_DSN = ''
+
+function withSentry (Child) {
+  return class WrappedComponent extends React.Component {
+    static getInitialProps (context) {
+      if (Child.getInitialProps) {
+        return Child.getInitialProps(context)
+      }
+      return {}
+    }
+    constructor (props) {
+      super(props)
+      this.state = {
+        error: null
+      }
+      Raven.config(
+        SENTRY_DSN
+      ).install()
+    }
+
+    componentDidCatch (error, errorInfo) {
```

Comment:
I wonder if this breaks Next's error handling / development workflow

Reply:
Why is that? I mean, `componentDidCatch` is new in React 16, if next.js is affected by using this method then we should review how next catches errors.

Reply:
Since we render the error page using componentDidCatch in some cases. We might have to `throw error` to make it bubble up after passing it to Sentry. But I'm not 100% sure 😄

Reply:
@timneutkens if that's the case Next.js must expose a way to do something with the error

Reply:
@sergiodxa yeah there are plans to add something similar to that into the core. 👍